todos/src/App.jsx

```diff
@@ -1,10 +1,65 @@
-import "./App.css";
+import Todo from "./components/Todo";
+
+import { faCirclePlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useState } from "react";
 
 function App() {
+  const [todos, setTodos] = useState([]);
+  const [newTodo, setNewTodo] = useState("");
+
+  const addNewTodo = () => {
+    if (newTodo.trim() != "") {
+      setTodos([
+        ...todos,
+        { id: todos.length + 1, todo: newTodo, hasEnded: false },
+      ]);
```

Comment:
Here you tried to update todos depending on the length of the previous todos, it is recommended to use a callback function which gets the previous state instead of directly using the state which might be outdated in some cases...